integration_tests/tests/integration_tests/ops_purge.rs

```diff
@@ -314,3 +437,155 @@ async fn test_purging_mints() {
+            r#"SELECT "cl_audits_v2"."tx", "cl_audits_v2"."leaf_idx", "cl_audits_v2"."tree" FROM "cl_audits_v2" LIMIT $1 OFFSET $2"#,
+            vec![10u64.into(), 0u64.into()],
+        ),
+        Transaction::from_sql_and_values(
+            DbBackend::Postgres,
+            r#"SELECT "cl_audits_v2"."tx", "cl_audits_v2"."leaf_idx", "cl_audits_v2"."tree" FROM "cl_audits_v2" LIMIT $1 OFFSET $2"#,
+            vec![10u64.into(), 10u64.into()],
+        ),
+    ];
+
+    assert_eq!(
+        expected_transactions,
+        db.connection().into_transaction_log()
+    );
+    assert!(res.is_ok());
+}
+
+#[tokio::test]
+#[serial]
+async fn test_purging_cnft_with_failed_transaction() {
+    init_logger();
+
+    let mock_asset_model = asset_model();
+    let mock_cl_audit_v2_model =
+        cl_audit_v2_model_with_tree(mock_asset_model.tree_id.clone().unwrap());
+
+    let mock_db = MockDatabase::new(DatabaseBackend::Postgres)
+        .append_query_results(vec![vec![mock_cl_audit_v2_model.clone()]])
+        .append_exec_results(vec![MockExecResult {
+            rows_affected: 1,
+            last_insert_id: 1,
+        }])
+        .append_query_results(vec![vec![mock_asset_model]])
+        .append_exec_results(vec![MockExecResult {
+            rows_affected: 1,
+            last_insert_id: 1,
+        }]);
+
+    let db = Arc::new(MockDatabasePool::from(mock_db));
+
+    let mut rpc_mock_responses = HashMap::new();
+    let rpc_response = encoded_confirmed_transaction_with_status_meta(
+        Some(TransactionError::InvalidAccountForFee),
+        Signature::try_from(mock_cl_audit_v2_model.tx.as_slice()).unwrap(),
+    );
+
+    rpc_mock_responses.insert(
+        RpcRequest::GetTransaction,
+        serde_json::json!(Response {
+            context: RpcResponseContext {
+                slot: 0,
+                api_version: Some(RpcApiVersion::default()),
+            },
+            value: json!(rpc_response),
+        }),
+    );
+
+    let rpc = Rpc::from_mocks(rpc_mock_responses, "succeeds".to_string());
+
+    let res = start_cnft_purge(
+        CnftArgs {
+            only_trees: None,
+            purge_args: Args {
+                purge_worker_count: 10,
+                mark_deletion_worker_count: 10,
+                batch_size: 10,
+                paginate_channel_size: 10,
+            },
+        },
+        Arc::clone(&db),
+        rpc,
+    )
+    .await;
+
+    let expected_db_txs = vec![
+        Transaction::from_sql_and_values(
+            DbBackend::Postgres,
+            r#"SELECT "cl_audits_v2"."tx", "cl_audits_v2"."leaf_idx", "cl_audits_v2"."tree" FROM "cl_audits_v2" LIMIT $1 OFFSET $2"#,
+            vec![10u64.into(), 0u64.into()],
+        ),
+        Transaction::from_sql_and_values(
+            DbBackend::Postgres,
+            r#"SELECT "cl_audits_v2"."tx", "cl_audits_v2"."leaf_idx", "cl_audits_v2"."tree" FROM "cl_audits_v2" LIMIT $1 OFFSET $2"#,
+            vec![10u64.into(), 10u64.into()],
+        ),
```

Comment:
These need to be updated no? Its on'y doing select. I'd expect some delete queries to be logged.